packages/suite/src/components/suite/layouts/WelcomeLayout/WelcomeSidebar.tsx

```diff
@@ -0,0 +1,155 @@
+import { Column, IconButton, variables } from '@trezor/components';
+import { setDiscreetMode } from 'src/actions/settings/walletSettingsActions';
+import { goto } from 'src/actions/suite/routerActions';
+import { useDispatch, useSelector, useTranslation } from 'src/hooks/suite';
+import { selectIsDiscreteModeActive } from 'src/reducers/wallet/settingsReducer';
+import { TrafficLightOffset } from '../../TrafficLightOffset';
+import styled from 'styled-components';
+import { borders, spacingsPx } from '@trezor/theme';
+import { Translation } from '../../Translation';
+import { useCallback, useEffect, useState } from 'react';
+import { SettingsAnchor } from 'src/constants/suite/anchors';
+import { selectTorState } from 'src/reducers/suite/suiteReducer';
+import { useLocation } from 'react-router-dom';
+import { ButtonVariant } from '@trezor/components/src/components/buttons/buttonStyleUtils';
+import { removeSettingsFromUrl } from './utils';
+import { TorContainerWithCheckIcon } from '../SuiteLayout/Sidebar/TorContainerWithCheckIcon';
+
+type ActivePage = 'trezor' | 'settings';
+
+const WELCOME_SIDEBAR_WIDTH = 84;
+const WELCOME_ICON_SIZE = 40;
+
+const WelcomeWrapper = styled.div`
+    background-color: ${({ theme }) => theme.backgroundSurfaceElevationNegative};
+    border-right: 1px solid ${({ theme }) => theme.borderElevation0};
+
+    @media (max-width: ${variables.SCREEN_SIZE.MD}) {
+        display: none;
+    }
+`;
+
+const LeftContainer = styled.div`
+    padding: ${spacingsPx.xs};
+    display: flex;
+    flex-direction: column;
+    justify-content: space-between;
+    height: 100%;
+`;
+
+const LeftWrapper = styled.div`
+    display: flex;
+    flex-direction: column;
+    gap: ${spacingsPx.xs};
+`;
+
+type ExtraIconButtonProps = {
+    variant: ButtonVariant;
+    borderRadius: string;
+    width: number;
+    height: number;
+};
+
+const extraIconButtonProps = (): ExtraIconButtonProps => {
+    return {
+        variant: 'welcome',
+        borderRadius: borders.radii.sm,
+        width: WELCOME_ICON_SIZE,
+        height: WELCOME_ICON_SIZE,
+    };
+};
+
+export const WelcomeSidebar = () => {
+    const dispatch = useDispatch();
+    const { translationString } = useTranslation();
+    const isDiscreetModeActive = useSelector(selectIsDiscreteModeActive);
+    const translationLabel = isDiscreetModeActive ? 'TR_SHOW_BALANCES' : 'TR_HIDE_BALANCES';
+    const { isTorLoading } = useSelector(selectTorState);
+    const [activePage, setActivePage] = useState<ActivePage>('trezor');
+    const location = useLocation();
+
+    const handleDiscreetModeClick = () => dispatch(setDiscreetMode(!isDiscreetModeActive));
+
+    const handleClickTrezor = () => {
+        setActivePage('trezor');
+        dispatch(goto('suite-start'));
+        const updatedUrl = removeSettingsFromUrl(window.location.pathname);
+        window.location.pathname = updatedUrl;
+        location.pathname = removeSettingsFromUrl(location.pathname);
```

Comment:
I think we shouldn't handle pathname this way. It's potentially dangerous and hard to understand why we do it. Couldn't we switch pages according to `activePage`?

Reply:
This is because of we are working with the foreground app here and the url isn't updated accordingly, We must do it manually to make things work as expected. Or to get rid of the foreground app. Or I can omit these lines but it's not gonna work well for some scenarios, if we don't find some other solution.

Working with `window` here is mainly for making it all work for web Suite also. Otherwise we would be fine with using just `useLocation` hook.